uber/models.py

```diff
@@ -206,10 +211,14 @@ def addons(self):
         """
         return []
 
-    @property
-    def cost_property_names(self):
+    @cached_classproperty
+    def cost_property_names(cls):
         """Returns the names of all cost properties on this model."""
-        return [name for name, attr in self._class_attrs.items() if isinstance(attr, cost_property)]
+        return [s for s in cls._class_attr_names if s != 'cost_property_names' and isinstance(getattr(cls, s), cost_property)]
+
+    @cached_classproperty
+    def multichoice_columns(cls):
```

Comment:
What's the functional difference between this and `all_checkgroups`? Is it that they return different types of things? I can see that `all_checkgroups` returns just the name -- I'm not super sure what it means to return a column instead.

We might want to name these similar things, although given the choice I'd prefer this kind of naming scheme (describing what it _is_) rather than the 'checkgroups' name which describes more how it's used in an HTML form.